setupext/platform.py

```diff
@@ -20,79 +20,102 @@
 import setupext
 import os
 import sys
+import sysconfig
+import distutils.log
 
 # This handles all of the work to make our platform specific extension options.
 
-platform_specific = {
-    'include_dirs': [
-        os.path.join('native', 'common', 'include'),
-        os.path.join('native', 'python', 'include'),
-        os.path.join('build', 'src'),
-    ],
-    'sources': [
-        os.path.join('build', 'src', 'jp_thunk.cpp')
-    ] + setupext.utils.find_sources(),
-}
-
-fallback_jni = os.path.join('native', 'jni_include')
-# try to include JNI first from eventually given JAVA_HOME, then from distributed
-java_home = os.getenv('JAVA_HOME', '')
-found_jni = False
-if os.path.exists(java_home) and sys.platform != "cygwin":
-    platform_specific['include_dirs'] += [os.path.join(java_home, 'include')]
-
-    # check if jni.h can be found
-    for d in platform_specific['include_dirs']:
-        if os.path.exists(os.path.join(d, 'jni.h')):
-            print("Found native jni.h at %s" % d)
-            found_jni = True
-            break
+
+def Platform(include_dirs=[], sources=[]):
```

Comment:
I vote against monkey-patching sys.platform and add an additional argument to this function, so we can pass the platform name from setup.py and just change it case the android argv is present. Sorry for being pedantic about this, but I've experienced horrible side-effects in the past, when changing such things (especially with distutils).

Reply:
Okay fine with me.   Not sure why the cross compiler doesn't have any symbols that tell you which you are compiling but I didn't find any.